subprojects/model-core/src/main/java/org/gradle/internal/instantiation/generator/AbstractClassGenerator.java

```diff
@@ -620,7 +620,11 @@ public String getName() {
         }
 
         public boolean isReadOnly() {
-            return mainGetter != null && setters.isEmpty();
+            return isReadable() && !isWritable();
+        }
+
+        public boolean isReadableAndNotDirectlyWritable() {
```

Comment:
```suggestion
        public boolean isReadableWithoutSetterOfPropertyType() {
```

How about ☝️ ? Do you think this is better?

Reply:
it definitely tell more, will use that

Reply:
renamed